src/executables/util_cs.c

```diff
@@ -3993,3 +3993,255 @@ tde (UTIL_FUNCTION_ARG * arg)
 error_exit:
   return EXIT_FAILURE;
 }
+
+static time_t
+parse_date_string_to_time (char *date_string)
+{
+  assert (date_string != NULL);
+
+  time_t result = 0;
+  struct tm time_data = { };
+
+  if (sscanf
+      (date_string, "%d-%d-%d:%d:%d:%d", &time_data.tm_mday, &time_data.tm_mon, &time_data.tm_year, &time_data.tm_hour,
+       &time_data.tm_min, &time_data.tm_sec) != 6)
+    {
+      return 0;
+    }
+
+  if (time_data.tm_mday < 1 || time_data.tm_mday > 31
+      || time_data.tm_mon < 1 || time_data.tm_mon > 12
+      || time_data.tm_year < 1900
+      || time_data.tm_hour < 0 || time_data.tm_hour > 23
+      || time_data.tm_min < 0 || time_data.tm_min > 59 || time_data.tm_sec < 0 || time_data.tm_sec > 59)
+    {
+      return 0;
+    }
+
+  time_data.tm_mon -= 1;
+  time_data.tm_year -= 1900;
+  time_data.tm_isdst = -1;
+
+  result = mktime (&time_data);
+
+  return result < 0 ? 0 : result;
+}
+
+int
+flashback (UTIL_FUNCTION_ARG * arg)
+{
+  UTIL_ARG_MAP *arg_map = arg->arg_map;
+  char er_msg_file[PATH_MAX];
+
+  const char *database_name = NULL;
+  int num_tables = 0;
+  dynamic_array *darray = NULL;
+  int i = 0;
+  char table_name_buf[SM_MAX_IDENTIFIER_LENGTH];
+  char *table_name;
+
+  const char *output_file = NULL;
+  FILE *outfp = NULL;
+  const char *user = NULL;
+  const char *dba_password = NULL;
+
+  char *start_date = NULL;
+  char *end_date = NULL;
+  time_t start_time = 0;
+  time_t end_time = 0;
+
+  bool is_detail = false;
+  bool is_oldest = false;
+
+  char *passbuf = NULL;
+  int error = NO_ERROR;
+
+  num_tables = utility_get_option_string_table_size (arg_map) - 1;
+  if (num_tables < 1)
+    {
+      /* TODO : error message will be dealt in other issue. Temporarily used fprintf */
+      fprintf (stderr, "too less arguments, dbname and table list are required\n");
+      goto print_flashback_usage;
+    }
+
+  database_name = utility_get_option_string_value (arg_map, OPTION_STRING_TABLE, 0);
+  if (database_name == NULL)
+    {
+      fprintf (stderr, "no database name\n");
+      goto print_flashback_usage;
+    }
+
+  output_file = utility_get_option_string_value (arg_map, FLASHBACK_OUTPUT_S, 0);
+  user = utility_get_option_string_value (arg_map, FLASHBACK_USER_S, 0);
+  dba_password = utility_get_option_string_value (arg_map, FLASHBACK_DBA_PASSWORD_S, 0);
+  start_date = utility_get_option_string_value (arg_map, FLASHBACK_START_DATE_S, 0);
+  end_date = utility_get_option_string_value (arg_map, FLASHBACK_END_DATE_S, 0);
+  is_detail = utility_get_option_bool_value (arg_map, FLASHBACK_DETAIL_S);
+  is_oldest = utility_get_option_bool_value (arg_map, FLASHBACK_OLDEST_S);
+
+  if (check_database_name (database_name))
```

Comment:
Yes, I thought it would be more readable but I did it for a consistency with legacy.
But I will move it, because this function just check the name of the database.